utils/build.ps1

```diff
@@ -1465,6 +1469,57 @@ function Build-Compilers() {
   }
 }
 
+# Reference: https://github.com/microsoft/mimalloc/tree/dev/bin#minject
+# TODO: Add ARM64
+function Build-Mimalloc() {
+  [CmdletBinding(PositionalBinding = $false)]
+  param
+  (
+    [Parameter(Position = 0, Mandatory = $true)]
+    [hashtable]$Arch
+  )
+
+  if ($Arch -eq $ArchX64) {
```

Comment:
Can we abort if a different architecture is passed? Or is this being allowed for some other reason? As this currently stands, it feels like we could silently fail to enable something :(.

Reply:
I think if we do that, the cross-compiling CI would likely fail

Reply:
Hmm, why would it fail? The check should be for what host we are compiling for, not what the build is right?

Reply:
Yeah, I was probably thinking if the mimalloc flag were to be enabled by default.

Done.